alacritty/src/input.rs

```diff
@@ -368,19 +368,36 @@ impl<'a, T: EventListener, A: ActionContext<T>> Processor<'a, T, A> {
 
     pub fn normal_mouse_report(&mut self, button: u8) {
         let (line, column) = (self.ctx.mouse().line, self.ctx.mouse().column);
+        let utf8 = self.ctx.terminal().mode().contains(TermMode::UTF8_MOUSE);
 
-        if line < Line(223) && column < Column(223) {
-            let msg = vec![
-                b'\x1b',
-                b'[',
-                b'M',
-                32 + button,
-                32 + 1 + column.0 as u8,
-                32 + 1 + line.0 as u8,
-            ];
-
-            self.ctx.write_to_pty(msg);
+        let max_point = if utf8 { 2015 } else { 223 };
+
+        if line >= Line(max_point) || column >= Column(max_point) {
+            return;
+        }
+
+        let mut msg = vec![b'\x1b', b'[', b'M', 32 + button];
+
+        let mouse_pos_encode = |pos: usize| -> Vec<u8> {
+            let pos = 32 + 1 + pos;
+            let first = 0xC0 + pos / 64;
+            let second = 0x80 + (pos & 0x3F);
```

Comment:
I think if we replace `0x3F` with `63` here, things suddenly start making much more sense. Like that it should be possible to see the correlation between the different statements and what they're supposed to do.

I don't think there's much value in changing `0xC0` and `0x80`, those values seem mostly arbitrary to me.